cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java

```diff
@@ -59,14 +60,16 @@ public MapReduceMetricsWriter(Job jobConf, BasicMapReduceContext context) {
   }
 
   public void reportStats() throws IOException, InterruptedException {
-    reportMapredStats();
-    reportSystemStats();
+    Counters jobCounters = jobConf.getCounters();
+    reportMapredStats(jobCounters);
+    reportSystemStats(jobCounters);
   }
 
   // job level stats from counters built in to mapreduce
-  private void reportMapredStats() throws IOException, InterruptedException {
+  private void reportMapredStats(Counters jobCounters) throws IOException, InterruptedException {
+    JobStatus jobStatus = jobConf.getStatus();
```

Comment:
Yes, that's what I also did in this PR.
I feel that each time we call `reportStats`, it should just report the stats as if it were a snapshot (as opposed to calling `job.getStatus` multiple times throughout the reporting iteration.